.github/workflows/reset-environment.yml

```diff
@@ -0,0 +1,219 @@
+name: Reset Environment
+
+on:
+  workflow_call:
+    secrets:
+      AZURE_APP_SERVICE_RESET_SERVICE_PRINCIPAL:
+        required: true
+    inputs:
+      # This needs to be stringified JSON because inputs don't support arrays, see 
```

Comment:
Does it make sense to run this on multiple machines at the same time with the same parameters? Because it doesn't seem so, and if not, then this workflow doesn't need to support a matrix. Same with `deploy-to-azure-app-service`.